Add LanceDB support (AP-411) and helper maintenance scripts

awilfox opened the pull request:
* Use LanceDB as the vector store.  Configured by `.env`.
  * This also means there is only a single mention of OllamaEmbeddings now.  This makes it easier to change that later.
* Centralise creation of the LanceDB object into the config module.  (This goes towards an eventual refactor of configuration, too.)
* Create scripts for fetching from TIND and running the pipeline manually, since the Web/CLI frontends don't run it automatically any more.

anarchivist reviewed (approved):
looks great, in general. wondering about a couple things, which are not at all blockers, and both related to docker/compose:

*  could we perhaps add a volume mapping in the compose file for `.data/lancedb:/tmp/lancedb-path`? 
* more broadly wondering if we can update the `ENTRYPOINT` for the app to be `/venv/bin/python` and have the `CMD` launch chainlit? that would allow one to, say, `docker compose run app -m willa.etl.pipeline`. 

if these should be separate, that's fine.

awilfox commented:
> * could we perhaps add a volume mapping in the compose file for `.data/lancedb:/tmp/lancedb-path`?

Added, and updated the documentation to reflect the default mounting path.

> * more broadly wondering if we can update the `ENTRYPOINT` for the app to be `/venv/bin/python` and have the `CMD` launch chainlit? that would allow one to, say, `docker compose run app -m willa.etl.pipeline`.

Changed.

I've also updated the `.dockerignore` file to ignore the storage paths by default.
